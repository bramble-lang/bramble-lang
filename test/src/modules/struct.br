fn my_main() {
    let ms: my_mod::MyStruct := my_mod::MyStruct{a: 5};
    let ms2: my_mod::inner::MyStruct := my_mod::inner::MyStruct{b: 10};

    printiln ms.a;
    printiln ms2.b;

    my_mod::test();

    return;
}

mod my_mod {
    struct MyStruct {
        a: i32,
    }

    mod inner {
        struct MyStruct {
            b: i32,
        }
    }

    fn test() {
        let ms: self::MyStruct := self::MyStruct{a: 11};
        let ms2: self::inner::MyStruct := inner::MyStruct{b: 33};

        printiln ms.a;
        printiln ms2.b;
        
        return;
    }
}