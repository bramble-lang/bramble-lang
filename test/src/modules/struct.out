5
10
11
33
